src/vs/code/electron-main/menus.ts

```diff
@@ -511,7 +511,8 @@ export class VSCodeMenu {
 
 		if (folders.length || files.length) {
 			openRecentMenu.append(__separator__());
-			openRecentMenu.append(new MenuItem(this.likeAction('clearRecentlyOpened', { label: mnemonicLabel(nls.localize({ key: 'miClearItems', comment: ['&& denotes a mnemonic'] }, "&&Clear Items")), click: () => this.windowsService.clearRecentPathsList() }, false)));
+			//workbench.action.clearRecentItems
```

Comment:
It means that you have left a commented out line `//workbench.action.clearRecentItems`, just remove it. 🙂